gcp-observability/build.gradle

```diff
@@ -29,27 +29,42 @@ dependencies {
     //  in gradle/libs.versions instead
     implementation project(':grpc-protobuf'),
             project(':grpc-stub'),
-            project(':grpc-alts'),
             project(':grpc-census'),
-            ("com.google.cloud:google-cloud-logging:${cloudLoggingVersion}"),
-            libraries.opencensus.contrib.grpc.metrics,
-            libraries.opencensus.exporter.stats.stackdriver,
-            libraries.opencensus.exporter.trace.stackdriver,
-            project(':grpc-xds'), // Align grpc versions
+            libraries.opencensus.contrib.grpc.metrics
+    // Avoid gradle using project dependencies without configuration: shadow
+    implementation ("com.google.cloud:google-cloud-logging:${cloudLoggingVersion}") {
+        exclude group: 'io.grpc', module: 'grpc-alts'
+        exclude group: 'io.grpc', module: 'grpc-netty-shaded'
+        exclude group: 'io.grpc', module: 'grpc-xds'
+    }
+    implementation (libraries.opencensus.exporter.stats.stackdriver) {
+        exclude group: 'io.grpc', module: 'grpc-alts'
+        exclude group: 'io.grpc', module: 'grpc-netty-shaded'
+        exclude group: 'io.grpc', module: 'grpc-xds'
+    }
+    implementation (libraries.opencensus.exporter.trace.stackdriver) {
+        exclude group: 'io.grpc', module: 'grpc-alts'
+        exclude group: 'io.grpc', module: 'grpc-netty-shaded'
+        exclude group: 'io.grpc', module: 'grpc-xds'
+    }
+
+    runtimeOnly libraries.opencensus.impl,
+            project(path: ':grpc-netty-shaded', configuration: 'shadow'),
+            project(path: ':grpc-xds', configuration: 'shadow'),
+            project(path: ':grpc-alts', configuration: 'shadow'),
+            project(':grpc-auth'), // Align grpc versions
```

Comment:
Why do we need `grpc-auth` and `grpc-grpclb`?

Reply:
They are transitive depedencies of cloud-logging and the exporters, and also alts. Unfortunately, Maven will select the version from cloud-logging or exporters over the one in grpc-alts, so we need to re-define them here with the greater version number to avoid Maven downgrading them.